doc/cookbook/service_proxy.md

```diff
@@ -0,0 +1,138 @@
+# Service Proxy
+
+- [Service Proxy](#service-proxy)
+  - [Basic: Load Balance](#basic-load-balance)
+    - [Dynamic: Integration with Service Registry](#dynamic-integration-with-service-registry)
+    - [Zookeeper](#zookeeper)
+    - [Consul](#consul)
+    - [Eureka](#eureka)
+
+Easegress servers as a resvert proxy. It can easily integrate with main stream Service Registries.  
+
+
+## Basic: Load Balance
+
+```yaml
+name: pipeline-reverse-proxy
+kind: HTTPPipeline
+flow:
+  - filter: proxy
+filters:
+  - name: proxy
+    kind: Proxy
+    mainPool:
+      servers:
+      - url: http://127.0.0.1:9095
+      - url: http://127.0.0.1:9096
+      - url: http://127.0.0.1:9097
+      loadBalance:
+        policy: roundRobin
+```
+
+
+### Dynamic: Integration with Service Registry
+
+* We integrate `Proxy` with service registry such as [Consul](https://github.com/megaease/easegress/blob/main/doc/controllers.md#consulserviceregistry), [Etcd](https://github.com/megaease/easegress/blob/main/doc/controllers.md#etcdserviceregistry), [Zookeeper](https://github.com/megaease/easegress/blob/main/doc/controllers.md#zookeeperserviceregistry), [Eureka](https://github.com/megaease/easegress/blob/main/doc/controllers.md#eurekaserviceregistry). You need to create one of them to connect the external service registry. The service registry config takes higher priority than static servers. If the dynamic servers pulling failed, it will use static servers if there are.
+
+
+### Zookeeper
+
+1. First we need to create a ZookeeperServiceRegistry in Easegress 
+
+``` yaml
+kind: ZookeeperServiceRegistry
+name: zookeeper-001
+zkservices: [127.0.0.1:2181]
+prefix: /services
+conntimeout: 6s
+syncInterval: 10s
+```
+
+2. Create a pipeline and set its serviceRegisty field into `zookeeper-001` and it will look up the zookeeper configuration for the service named `springboot-application-order` as in filed `serviceName`.    
```

Comment:
```suggestion
2. Create a pipeline and set its `serviceRegisty` field into `zookeeper-001` and it will look up the zookeeper configuration for the service named `springboot-application-order` as in filed `serviceName`.    
```